### PR TITLE
Feature/story test/aave

### PR DESCRIPTION
**Pull request by dumebi**
- Implemented Aave tests
- fixed `call dapp failure` on sushiswap
- readjusted story test to fix 'TAKE_OUT` loan bug

### CODE REVIEW

File: test/helpers/story/drivers/dapp-story-test-driver.ts [hunk 1]
```diff
@@ -66,14 +71,26 @@ export default class DappStoryTestDriver extends StoryTestDriver {
...
+          let shouldPass = true
+          if (!loan) {
+            shouldPass = false
+          }
+          if (shouldPass) {
+            // await LoanStoryTestDriver.takeOutLoan(hre, {})
+            await DappStoryTestDriver.lendAave(hre, loan)
+          } else {
+            await DappStoryTestDriver.lendAave(hre, loan)
+          }
```
Comment (passabilities) [inline test/helpers/story/drivers/dapp-story-test-driver.ts:87]: why are these the same for both scenarios?

File: test/helpers/story/drivers/dapp-story-test-driver.ts [hunk 2]
```diff
@@ -132,40 +149,45 @@ export default class DappStoryTestDriver extends StoryTestDriver {
...
           } else {
-            await DappStoryTestDriver.swapSushiSwap(hre, loan).catch((error) => {
-              expect(error).to.exist
-            })
+            await DappStoryTestDriver.swapSushiSwap(hre, loan).catch(
+              (error) => {
+                expect(error).to.exist
+              }
+            )
           }
```
Comment (passabilities) [inline test/helpers/story/drivers/dapp-story-test-driver.ts:189]: when are we testing `shouldPass === true` ?

File: test/helpers/story/drivers/dapp-story-test-driver.ts [hunk 3]
```diff
@@ -174,7 +196,62 @@ export default class DappStoryTestDriver extends StoryTestDriver {
...
+        details.loan.id,
+        details.loan.lendingToken,
+        details.loan.borrowedAmount
+      )
+
+    const escrowAddress = await diamond.getLoanEscrow(details.loan.id)
+
+    const aDaiBalance = await aToken.balanceOf(escrowAddress)
+
+    aDaiBalance.eq(0).should.eql(false, '')
```
Comment (passabilities) [inline test/helpers/story/drivers/dapp-story-test-driver.ts:223]: For Aave, the tokens are a 1:1 ratio. We should instead test that the `aToken` balance is equal to the amount deposited

File: test/helpers/story/drivers/dapp-story-test-driver.ts [hunk 4]
```diff
@@ -174,7 +196,62 @@ export default class DappStoryTestDriver extends StoryTestDriver {
...
+      .connect(details.borrower.signer)
+      .aaveWithdrawAll(details.loan.id, details.lendingToken.address)
+
+    const escrowAddress = await diamond.getLoanEscrow(details.loan.id)
+
+    const tokenAddresses = await diamond.getEscrowTokens(details.loan.id)
+    tokenAddresses.should.not.include(aToken.address)
+
+    const aDaiBalance = await aToken.balanceOf(escrowAddress)
+    aDaiBalance.eq(0).should.eql(true, '')
```
Comment (passabilities) [inline test/helpers/story/drivers/dapp-story-test-driver.ts:248]: again, we should test that the resulting amount withdrawn is equal to the balance of the `aToken`

**Review by passabilities — CHANGES_REQUESTED** [review]

File: test/helpers/story/drivers/dapp-story-test-driver.ts (hunk 4 above)
Comment (dumebi) [reply test/helpers/story/drivers/dapp-story-test-driver.ts:248]: this is a withdrawAll

File: test/helpers/story/drivers/dapp-story-test-driver.ts (hunk 1 above)
Comment (dumebi) [reply test/helpers/story/drivers/dapp-story-test-driver.ts:87]: fixed

File: test/helpers/story/drivers/dapp-story-test-driver.ts (hunk 2 above)
Comment (dumebi) [reply test/helpers/story/drivers/dapp-story-test-driver.ts:189]: i want to merge this into currently existing sushiswap  branch

File: test/helpers/story/drivers/dapp-story-test-driver.ts [hunk 5]
```diff
@@ -132,40 +150,45 @@ export default class DappStoryTestDriver extends StoryTestDriver {
...
         break
       }
       case 'SUSHISWAP': {
-        const newTest = new Test('SUSHISWAP Swap DAPP', (async () => {
+        const newTest = new Test('SUSHISWAP Swap DAPP', async () => {
           const borrower = await getNamedSigner('borrower')
           const loan = await LoanStoryTestDriver.getLoan(hre, borrower)
-          let shouldPass = true
+          let shouldPass = false
           //read the state and determine if this should pass
```
Comment (sypeer) [inline test/helpers/story/drivers/dapp-story-test-driver.ts:174]: Thoughts on adding descriptive comments like this over key steps in the story?

**Review by sypeer — APPROVED** [review]

File: test/helpers/story/drivers/dapp-story-test-driver.ts (hunk 5 above)
Comment (dumebi) [reply test/helpers/story/drivers/dapp-story-test-driver.ts:174]: okay will do

**Review by mkhoury93 — APPROVED** [review]

**Pull request merged**